text/0000-sbom.md

```diff
@@ -0,0 +1,126 @@
+- SWID: SWID is a formal industry standard used by various commercial software publishers designed with software inventory and entitlements management in mind.
+
+# Motivation
+[motivation]: #motivation
+
+## Why should we do this?
+
+Quoting [NTIA Whitepaper][NTIA-WP] - 
+> Modern software systems involve increasingly complex and dynamic supply chains. Unfortunately, the composition and functionality of these systems lacks transparency; this contributes substantially to cybersecurity risks, alongside the cost of development, procurement, and maintenance. This has broad implications in our interconnected world; risk and cost affect collective goods, like public safety and national security, in addition to the products and services upon which businesses rely.
+> We propose that increased supply chain transparency through SBOMs can reduce cybersecurity risks and overall costs by:
+> - Enhancing the identification of vulnerable systems and the root cause of incidents
+> - Reducing unplanned and unproductive work
+> - Supporting more informed market differentiation and component selection
+> - Reducing duplication of effort by standardizing formats across multiple sectors
+> - Identifying suspicious or counterfeit software components
+
+Currently since container images built by Cloud Native Buildpacks don't follow a generic file system layout, they may not be easily scannable by container scanning tools. We do however provide a BOM which should greatly help sidestep the entire manual scanning process and speed up things like CVE detection by directly providing the CVE scanner with a BOM.
+
+However, since we currently do not impose any standard for what the BOM should look like and since the metadata table in BOM is a freeform table, it is very hard to have consistent BOMs that can be used by CVE scanners.
+
+Having a structured and consistent SBOM that can be consumed by downstream systems could greatly enhance the security proposition around buildpacks.
+
+## What use cases does it support?
+
+- Consistent SBOM across buildpacks
+- Improved integration with other supply chain analysis and scanning tools
+
+## What is the expected outcome?
+
+- Use of CycloneDX as the SBOM storage format and updates the the specification and lifecycle to reflect the same.
+- Moving the SBOM from `<layer>.toml`, `launch.toml` and `build.toml` to a separate file.
+- Enhancements to the platform spec and `pack` to allow easy interaction with SBOMs including conversion between major SBOM formats.
+
+# What it is
+[what-it-is]: #what-it-is
+
+Currently the SBOM is defined in the `<layer>.toml`, `launch.toml` and `build.toml` respectively under the `[bom]` table.
+It may not be the most user-friendly way for buildpack authors to create SBOM documents in the above format. This RFC proposes that the `bom` be moved to `<layer>.bom.<ext>`, `launch.bom.<ext>` and `build.bom.<ext>` instead where `<ext>` will be `cdx.json`
```

Comment:
What is the reason for even creating the separate files for `layer`, `launch`, and `build` if we are just going to merge them together into one file down the line? 
Is there any intent here to have separate files so that users can also access the separate BOMs?

Reply:
One benefit of keeping them separate: the lifecycle could automatically insert metadata into each CycloneDX SBoM entry that maps the component to a layer digest and buildpack (and/or marks the component as build-only).

Reply:
Apart from what @sclevine said, each layer in a buildpack could have a different lifecycle (some could be launch only etc.). Keeping BOMs separate is an artifact of tying them to a similar lifecycle as their layers so that the BOMs and the contents they are describing don't drift apart. As for separate BOMs for diffirent buildpacks - It would get difficult for BP authors to write to a shared BOM file and random buildpacks could potentially modify BOMs for other buildpacks.

Reply:
I think that your lifecycle description makes sense for the `build` and `launch` BOM files. I am still confused by the presence of the `layer` BOM file. What purpose does it serve and how would it be utilized?

Reply:
What goes in the `layer` BOM that does not go into `build` or `launch` BOM? I understand the separation in terms of the "lifecycles" but I am unclear where just the `layer` BOM comes in.

Reply:
Some details and BG are in this RFC - https://github.com/buildpacks/rfcs/blob/main/text/0087-bom-in-layer-metadata.md

The idea was for the build and launch BOM to include changes/contribution to the app/workspace and the layer bom to have the layer specific bom.

Reply:
> The buildpack SHOULD NOT add bom to a layer content metadata file describing the contents that are not contributed by the same layer. BOM entries that describe contribution to the application directory should go in the buildpack-level BOM entries in launch.toml or build.toml.